util/build_configs/cray-internal/chapel_package-cray.bash

```diff
@@ -191,16 +191,57 @@ log_debug "Generate modulefile-$pkg_version ..."
 $cwd/generate-modulefile.bash > "$rpmbuild_dir/modulefile-$pkg_version"
 chmod 644 "$rpmbuild_dir/modulefile-$pkg_version"
 
+# Generate Lua modulefile for PE Lmod Hierarchy.
+
+log_debug "Generate Lua modulefile ..."
+
+(
+    if [ "$chpl_platform" = hpe-cray-ex ]; then
+        platform_prefix=/opt/cray
+    else
+        platform_prefix=/opt
+    fi
+    $cwd/process-template.py pkg_version="$pkg_version" \
+                             platform_prefix="$platform_prefix" \
+        --template $cwd/chapel.modulefile.lua.template \
+        --output $rpmbuild_dir/modulefile-lua-$pkg_version
+    chmod 644 $rpmbuild_dir/modulefile-lua-$pkg_version
+)
+
 # Generate set_default script, w versions
 
 log_debug "Generate set_default_chapel_$pkg_version ..."
 $cwd/generate-set_default.bash > "$rpmbuild_dir/set_default_chapel_$pkg_version"
 chmod 755 "$rpmbuild_dir/set_default_chapel_$pkg_version"
 
-# Generate chapel.spec, w versions
+# Generate RPM spec file.
 
 log_debug "Generate chapel.spec ..."
-$cwd/generate-rpmspec.bash > "$rpmbuild_dir/chapel.spec"
+
+(
+    if [ "$chpl_platform" = hpe-cray-ex ]; then
+        lmod_network=ofi
+        platform_prefix=/opt/cray
+        set_def_subdir=admin-pe/set_default_files
+    else
+        platform_prefix=/opt
+        set_def_subdir=cray/admin-pe/set_default_files
+    fi
+    lmod_prefix=/opt/cray/pe/lmod/modulefiles/mpi
+    lmod_suffix=${lmod_network}/1.0/cray-mpich/8.0
```

Comment:
Should we be concerned about hardcoding mpich versions here (and gnu/cce versions below)?

Reply:
These are least-common-denominator versions, defined by the Lmod Hierarchy itself.  See https://connect.us.cray.com/confluence/display/MCE/Lmod+Modules#LmodModules-ListofCompatibleVersions.